### PR TITLE
Add recognition of Distinct identifiers in grammar files

### LINKED ISSUE #4468: ignored distinct in scalar function
### Steps to reproduce the behavior (Required)
```
CREATE TABLE v0 ( v1 varchar(27) ) ; 
INSERT INTO v0 VALUES ( "10" ) ,( "1" ) ,( "1" ) ;
SELECT substr(DISTINCT v1 , 9223372036854775807 , ( v1 NOT IN ( 0 ) AND v1 NOT IN ( 0 ) ) ) FROM v0 ; 
```

### Expected behavior (Required)
Syntax error

### Real behavior (Required)

ignored distinct keyword

### StarRocks version (Required)
 - main 2.0.1 2.0.2

### LINKED ISSUE #4488: unknown error in aggregate STD
### Steps to reproduce the behavior (Required)
```
CREATE TABLE v0 ( v1 VARCHAR ( 55 ) ) ;
INSERT INTO v0 ( v1 ) VALUES ( 'x' ) , ( NULL ) , ( 'x' ) , ( NULL ) ;
SELECT STD ( DISTINCT v1 ) AS v2 FROM v0 GROUP BY v1 ;
```

### Expected behavior (Required)
query success

### Real behavior (Required)
Unknown error
```
java.lang.IllegalStateException: null
        at com.google.common.base.Preconditions.checkState(Preconditions.java:494) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.sql.optimizer.rule.transformation.SplitAggregateRule.implementTwoStageAgg(SplitAggregateRule.java:255) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.rule.transformation.SplitAggregateRule.transform(SplitAggregateRule.java:185) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.task.ApplyRuleTask.execute(ApplyRuleTask.java:62) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.task.SeriallyTaskScheduler.executeTasks(SeriallyTaskScheduler.java:42) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.Optimizer.memoOptimize(Optimizer.java:129) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.Optimizer.optimize(Optimizer.java:92) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.createQueryPlan(StatementPlanner.java:81) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:42) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:327) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:283) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:430) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:666) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:55) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_272]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_272]
```

### StarRocks version (Required)
 - main 6c90268eb

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4488 #4468

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java [hunk 1]
```diff
@@ -134,4 +147,34 @@ private void setOutfileSink(QueryStatement queryStmt, ExecPlan plan) {
         ResultSink resultSink = (ResultSink) topFragment.getSink();
         resultSink.setOutfileInfo(queryStmt.getOutFileClause());
     }
+
+    public static boolean supportedByNewParser(StatementBase statement) {
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java:151]: When could they converge?

**Review by murphyatwork — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 39 / 47 (82.98%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/ParsingException.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|32|38|84.21%|
|:large_blue_circle:|com/starrocks/sql/parser/SqlParser.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/StatementPlanner.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|3|3|100.00%|

**Review by murphyatwork — DISMISSED** [review]

**Review by Youngwb — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java [hunk 2]
```diff
@@ -134,4 +147,34 @@ private void setOutfileSink(QueryStatement queryStmt, ExecPlan plan) {
...
+                || statement instanceof CreateViewStmt
+                || statement instanceof InsertStmt
+                || statement instanceof QueryStmt
+                || statement instanceof QueryStatement
+                || statement instanceof ShowDbStmt
+                || statement instanceof ShowTableStmt
+                || statement instanceof UpdateStmt;
+    }
+
+    public static boolean supportedByNewAnalyzer(StatementBase statement) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java:163]: Would better make it to be a method  in StatementBase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 39 / 47 (82.98%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/ParsingException.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|32|38|84.21%|
|:large_blue_circle:|com/starrocks/sql/parser/SqlParser.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/analysis/StatementBase.java|2|2|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 39 / 47 (82.98%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/ParsingException.java|0|2|00.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|32|38|84.21%|
|:large_blue_circle:|com/starrocks/sql/parser/SqlParser.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/StatementPlanner.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|3|3|100.00%|

**Review by Youngwb — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**